pytest_asyncio/plugin.py

```diff
@@ -6,8 +6,36 @@
 import inspect
 import socket
 import warnings
+from typing import (
+    Any,
+    Awaitable,
+    Callable,
+    Dict,
+    Iterable,
+    Iterator,
+    List,
+    Optional,
+    Set,
+    TypeVar,
+    Union,
+    cast,
+    overload,
+)
 
 import pytest
+from typing_extensions import Literal
+
+_ScopeName = Literal["session", "package", "module", "class", "function"]
+_T = TypeVar("_T")
+
+FixtureFunction = TypeVar("FixtureFunction", bound=Callable[..., object])
```

Comment:
This is supposed to only be applied to async functions, right?

It should probably look like this: https://github.com/hynek/prometheus-async/pull/21/files#diff-df328b684166777b1785c7abc3acfcb1c1ec5c236567bc39472d1389726267a7R32

Maybe the ParamSpec is unnecessary, but we can verify the return type is a subtype of Awaitable.

Reply:
Good idea.
Actually, it can be either an Awaitable or AsyncIterator.
I thought about it but was too lazy.

Reply:
I understand completely. ;)

Reply:
Done.
`ParamSpec` cannot be applied as `Callable[ParamSpec, RetType]` argument in `mypy==0.931`.
The `ParamSpec` implementation is not full yet, unfortunately. I hope it will be fixed sooner or later, ParamSpec looks like the very useful thing.

Reply:
tricky